lib/framework/data_context.dart

```diff
@@ -50,15 +51,24 @@ import 'package:jsparser/jsparser.dart';
 
 /// manages Data and Invokables within the current data scope.
 /// This class can evaluate expressions based on the data scope
-class DataContext {
+class DataContext implements Context {
+  static const String globalContextKey = '__internal__parent__context__';
```

Comment:
I created the parent context as a key inside the context. Not my ideal solution but had to do it to make sure that our copy operations copy over the parent context as well